modules/RTC/JitsiLocalTrack.js

```diff
@@ -977,4 +977,55 @@ export default class JitsiLocalTrack extends JitsiTrack {
     unmute() {
         return this._queueSetMuted(false);
     }
+
+    /**
+     * Applies media constraints to the current MediaStreamTrack.
+     *
+     * @param {Object} constraints - Media constraints to apply.
+     * @returns {Promise<void>}
+     */
+    async applyConstraints(constraints) {
+        const initialSettings = this.track.getSettings();
+        const mediaType = this.getType();
+
+        this.stopStream();
+
+        const constraintsToApply = {
+            ...initialSettings,
+            ...constraints
+        };
+        let mediaStreamData;
+
+        try {
+            const mediaStreamsData
+                = await RTCUtils.obtainAudioAndVideoPermissions({
+                    devices: [ mediaType ],
+                    micDeviceId: constraintsToApply.deviceId,
+                    constraints: { [mediaType]: constraintsToApply }
+                });
+
+            mediaStreamData = mediaStreamsData[0];
+
+            if (!mediaStreamData) {
+                throw new Error('No media stream data');
+            }
+        } catch (error) {
+            this.stream.getTracks().forEach(track => {
+                if (track.start) {
+                    track.start();
+                }
+            });
+
+            return;
+        }
+
+        await this.conference._removeLocalTrackFromPc(this);
+
+        this._setStream(mediaStreamData.stream);
+        this.track = mediaStreamData.track;
```

Comment:
How will user code know the stream and track have been replaced?

Reply:
We only need to reattach the new MediaStream to the corresponding containers for video tracks(which we haven't tested yet) but other than that I don't see why this would be needed for audio tracks. This is how setEffect is also implemented.

Reply:
Right, not for audio tracks, but something is needed for video tracks? Also note that RN can't re-attach itself to the containers...

Reply:
Oh ok, lets restrict this to audio tracks for now so that Hugo can proceed with the rest of the work for the GSOC project?